cmake/kokkos_arch.cmake

```diff
@@ -588,32 +588,39 @@ IF (KOKKOS_ENABLE_SYCL)
 ENDIF()
 
 # Check support for device_global variables
-# FIXME_SYCL Once the feature test macro SYCL_EXT_ONEAPI_DEVICE_GLOBAL is
-#            available, use that instead.
+# FIXME_SYCL Even if SYCL_EXT_ONEAPI_DEVICE_GLOBAL is defined, we still can't
+#            use device global variables with shared libraries
```

Comment:
This is true for all the models except the host right? Or is there some extra restriction for sycl?

Reply:
I couldn't find much about that restriction in the SYCL specs and didn't get a clear answer from Intel. I'm happy to explore if we can now have a non-`SYCL_EXTERNAL`/rdc option (like we have for the other backends), which might allow the use of shared libraries. 
I only mentioned it specifically since some functionality needs shared libraries (and `ZES_ENABLE_SYSMAN=1`) like querying free and total device memory.

Reply:
When you say “needs shared libraries” what do you mean? As far as I know, none of the backends support having device code in one shared object call device code in another one.  Is it more than that, like code in one shared object can’t dereference an address from another? Sorry if I’m being obtuse, I’m trying to tease out what the specific limitation is so I can keep track of it, and thank you for the explanation so far.

Reply:
> When you say “needs shared libraries” what do you mean?

`KokkosKernels` needs/has/wants functionality for querying free and total device memory and the implementation requires `Kokkos` and `KokkosKernels` to be built as shared libraries (and setting `ZES_ENABLE_SYSMAN=1`). That is unrelated to how global device variables are used. That means that we can't support arbitrary-sized atomics and that memory query at the same time.

> As far as I know, none of the backends support having device code in one shared object call device code in another one. Is it more than that, like code in one shared object can’t dereference an address from another?

I don't know about more restrictions. I normally don't use `SYCL_EXTERNAL`/rdc much and I have only seen linker errors for this feature. Neither https://www.intel.com/content/www/us/en/developer/articles/technical/use-the-sycl-external-macro.html nor https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#subsec:syclexternal mention shared libraries in particular, but only talk about the need to use `SYCL_EXTERNAL` if device functions are used in a different translation unit.